plugins/inputs/vault/vault.go

```diff
@@ -0,0 +1,219 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Vault configuration object
+type Vault struct {
+	URL string `toml:"url"`
+
+	Token       string `toml:"token"`
+	TokenString string `toml:"token_string"`
+
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+
+	tls.ClientConfig
+
+	roundTripper http.RoundTripper
+}
+
+const timeLayout = "2006-01-02 15:04:05 -0700 MST"
+
+var sampleConfig = `
+  ## URL for the Vault agent
+  # url = "http://127.0.0.1:8200"
+
+  ## Use Vault token for authorization.
+  ## Vault token configuration is mandatory.
+  ## If both are empty or both are set, an error is thrown.
+  # token = "/path/to/auth/token"
+  ## OR
+  token_string = "s.CDDrgg5zPv5ssI0Z2P4qxJj2"
```

Comment:
Can we name is like
```suggestion
  # token_file = "/path/to/auth/token"
  ## OR
  token = "s.CDDrgg5zPv5ssI0Z2P4qxJj2"
```
as it makes things clearer and shorter IMO. Not a must though.